tests/tests/test_security.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python
+# Copyright 2016 Mender Software AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        https://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from fabric.api import *
+import pytest
+import time
+from common import *
+from common_docker import *
+from common_setup import *
+from helpers import Helpers
+from MenderAPI import auth, adm, deploy, image, logger
+from common_update import common_update_procedure
+from mendertesting import MenderTesting
+import subprocess
+import sys
+sys.path.insert(0, '..')
+import conftest
+
+
+class TestSecurity(MenderTesting):
+
+    @pytest.mark.usefixtures("standard_setup_without_client")
+    def test_ssl_only(self):
+        """ make sure we are not exposing any non-ssl connections"""
+
+        # get all exposed ports from docker
+        exposed_hosts = subprocess.check_output("docker ps | grep %s | grep -o -E '0.0.0.0:[0-9]*'" % (conftest.docker_compose_instance), shell=True)
```

Comment:
playing with python:

```
# make dict: service name -> ports, grab lines, split on ';', grab service name, split on '_', pick middle element
In [59]: d = {srv[0].split('_')[1]: srv[1] for srv in [l.split(';') for l in subprocess.check_output('docker ps --format "{{.Names}};{{.Ports}}"', shell=True).decode().splitlines()]}

In [60]: d
Out[60]: 
{'mender-api-gateway': '0.0.0.0:443->443/tcp',
 'mender-conductor': '8080/tcp',
 'mender-deployments': '8080/tcp',
 'mender-device-adm': '8080/tcp',
 'mender-device-auth': '8080/tcp',
 'mender-dynomite': '8101-8102/tcp, 22122/tcp, 22222/tcp',
 'mender-elasticsearch': '9200/tcp, 9300/tcp',
 'mender-gui': '80/tcp',
 'mender-inventory': '8080/tcp',
 'mender-useradm': '8080/tcp',
 'minio': '9000/tcp',
 'mongo-common': '27017/tcp',
 'storage-proxy': '0.0.0.0:9000->9000/tcp'}
# grab public ports
In [62]: re.findall(r'0.0.0.0:[0-9]*', ' '.join(d.values()))
Out[62]: ['0.0.0.0:9000', '0.0.0.0:443']
```
Don't know if you can have docker print only public ports